app/controllers/projects/queries_controller.rb

```diff
@@ -31,13 +31,17 @@ class Projects::QueriesController < ApplicationController
 
   # No need for a more specific authorization check. That is carried out in the contracts.
   before_action :require_login
-  before_action :find_query, only: %i[update destroy]
+  before_action :find_query, only: %i[show update destroy publish unpublish]
   before_action :build_query_or_deny_access, only: %i[new create]
 
   current_menu_item [:new, :create] do
     :projects
   end
 
+  def show
+    redirect_to projects_path(query_id: @query.id)
+  end
```

Comment:
I had a few cases, where the browser encountered an error on the POST to `/queries/1` and that then showed up as the URL in the browser, when refreshing I was met with a 404 error. Personally, I want to surprise the user the least, and if they refresh they should land on a page that makes sense and that what this redirect does

Reply:
I assume you mean resubmitting the url from the url bar, as refresh of post POST by browsers should still be a POST request ([discussion about it](https://github.com/whatwg/html/issues/6600)). Not sure it will reduce user surprise, as changes that were in post data will be gone